From 2028565ba5d63790d1d9b8d1d0ac629b4aab0a12 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Wed, 14 Oct 2026 16:58:32 -0700
Subject: [PATCH] Encode negative PSBT test versions through unsigned Python
 wire API

The current Python framework stores the raw uint32 wire representation. Mask signed fixture inputs while continuing to assert signed RPC decode results, preserving the negative-version ordering regression.

diff --git a/test/functional/dogecoin_rpc_psbt_versions.py b/test/functional/dogecoin_rpc_psbt_versions.py
index a4d99e9fe1e321e7a58a057333d1c95ab5195b07..c3b7027d29f10e5a8222cbe35e1149d6578eed6b 100644
--- a/test/functional/dogecoin_rpc_psbt_versions.py
+++ b/test/functional/dogecoin_rpc_psbt_versions.py
@@ -19,7 +19,7 @@ class DogecoinPSBTVersionsTest(BitcoinTestFramework):
 
         def psbt(version, prevhash):
             tx = CTransaction()
-            tx.nVersion = version
+            tx.nVersion = version & 0xFFFFFFFF
             tx.vin = [CTxIn(COutPoint(prevhash, 0))]
             tx.vout = [CTxOut(1000, CScript([OP_TRUE]))]
             return node.converttopsbt(tx.serialize().hex())
